rq/job.py

```diff
@@ -42,6 +42,21 @@ class JobStatus(str, Enum):
     CANCELED = 'canceled'
 
 
+class Dependency:
+    def __init__(self, jobs, allow_failure: bool = False):
+        jobs = ensure_list(jobs)
+        if not all(
+            isinstance(job, Job) or isinstance(job, str)
+            for job in jobs
+        ):
+            raise ValueError("jobs: must contain objects of type Job and/or strings representing Job ids")
+        elif len(jobs) < 1:
+            raise ValueError("jobs: cannot be empty.")
+
+        self.dependencies = jobs
+        self.allow_failure = allow_failure
```

Comment:
Can we make this into an enum to accommodate future variations? Some possible variations:
1. Execute immediately regardless of retries, or wait until retries are done
2. If a job has multiple dependencies, execute immediately after any one is successful3. 

Something like `Dependency(job, execute_when=EXECUTE_WHEN.SUCCESS)` or something like that (but with better naming possibly)?

Reply:
A few things that may come into consideration with regards to dependency logic:
1. If it has multiple dependencies
2. Retries

We don't need to address these in this PR, but preferably we design the dependency API so that it's more future proof.